src/components/views/elements/DirectorySearchBox.js

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2016 OpenMarket Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import classnames from 'classnames';
+
+export default class DirectorySearchBox extends React.Component {
+    constructor() {
+        super();
+        this.collectInput = this.collectInput.bind(this);
+        this.onClearClick = this.onClearClick.bind(this);
+        this.onChange = this.onChange.bind(this);
+        this.onKeyUp = this.onKeyUp.bind(this);
+        this.onJoinButtonClick = this.onJoinButtonClick.bind(this);
+
+        this.input = null;
+
+        this.state = {
+            value: '',
+        };
+    }
+
+    collectInput(e) {
+        this.input = e;
+    }
+
+    onClearClick() {
+        this.setState({value: ''});
+
+        if (this.input) {
+            this.input.focus();
+
+            if (this.props.onClear) {
+                this.props.onClear();
+            }
+        }
+    }
+
+    onChange(ev) {
+        if (!this.input) return;
+        this.setState({value: ev.target.value});
+
+        if (this.props.onChange) {
+            this.props.onChange(this.state.value);
+        }
+    }
+
+    onKeyUp(ev) {
+        if (ev.key == 'Enter') {
+            if (this.props.onJoinClick) {
+                this.props.onJoinClick(this.state.value);
+            }
+        }
+    }
+
+    onJoinButtonClick() {
+        if (this.props.onJoinClick) {
+            this.props.onJoinClick(this.state.value);
+        }
+    }
+
+    _contentLooksLikeAlias() {
```

Comment:
The logic, ftr, was treating functions passed to other bits (ie. the react callbacks) as public facing.